File: pilas.cs
using System;

class StackExample
{
    const int MAX_SIZE = 100;
    static int[] stack = new int[MAX_SIZE];
    static int top = -1;

    static void Push(int value)
    {
        if (top >= MAX_SIZE - 1)
        {
            Console.WriteLine("Stack Overflow");
            return;
        }
        stack[++top] = value;
    }

    static int Pop()
    {
        if (top < 0)
        {
            Console.WriteLine("Stack Underflow");
            return -1;
        }
        return stack[top--];
    }

    static int Peek()
    {
        if (top < 0)
        {
            Console.WriteLine("Stack is empty");
            return -1;
        }
        return stack[top];
    }

    static bool IsEmpty()
    {
        return top == -1;
    }

    static bool IsFull()
    {
        return top >= MAX_SIZE - 1;
    }

    static void Main()
    {
        Push(10);
        Push(20);
        Push(30);

        Console.WriteLine("Top element is: " + Peek());
        Console.WriteLine("Popped element is: " + Pop());
        Console.WriteLine("Is stack empty? " + (IsEmpty() ? "Yes" : "No"));
    }
}

File: pilas.cpp
MAX_SIZE = 100

stack = [0] * MAX_SIZE
top = -1

def push(value):
    global top
    if top >= MAX_SIZE - 1:
        print("Stack Overflow")
        return
    top += 1
    stack[top] = value

def pop():
    global top
    if top < 0:
        print("Stack Underflow")
        return -1
    value = stack[top]
    top -= 1
    return value

def peek():
    if top < 0:
        print("Stack is empty")
        return -1
    return stack[top]

def is_empty():
    return top == -1

def is_full():
    return top >= MAX_SIZE - 1

# Main
push(10)
push(20)
push(30)

print("Top element is:", peek())
print("Popped element is:", pop())
print("Is stack empty?", "Yes" if is_empty() else "No")
